### PR TITLE
Spot price optional for SIR, SFR

### LINKED ISSUE #3548: Spot price should be optional in aws_spot_instance_request resource
In version 0.11.3, spot_price is **required**. However, according to the [AWS docs](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_RequestSpotInstances.html), the spot price is not required and defaults to the On-Demand price. This is _very convenient_ because you don't need to provide any price if you are willing to have the On-Demand price be your maximum price. (Of course, you are hoping to come in below that and usually you will.)

### Affected Resource(s)
- aws_spot_instance_request
- aws_spot_fleet_request

### Expected Behavior
You don't provide a `spot_price` and AWS will default to the On-Demand price (max price, not price you are necessarily going to pay).

### Actual Behavior
Throws an error because `spot_price` is required.

### Steps to Reproduce
1. Create an aws_spot_instance_request/aws_spot_fleet_request without spot_price
2. When you `terraform init`, error is thrown

### PR DESCRIPTION
**Pull request by YakDriver**
Code + tests + docs
Fixes #3548, formerly #3549 (closed)

### CODE REVIEW

File: aws/resource_aws_spot_fleet_request_test.go [hunk 1]
```diff
@@ -1511,6 +1511,87 @@ resource "aws_spot_fleet_request" "foo" {
 `, rName, rInt, rInt, rName)
 }
 
+func testAccAWSSpotFleetRequestConfigWithoutSpotPrice(rName string, rInt int) string {
```
Comment (bflad) [inline aws/resource_aws_spot_fleet_request_test.go]: This PR is in great shape -- only item is that this new test configuration does not look like its being called. If you fix that up then we should be able to get this in 👍

**Comment by YakDriver** [comment]
@bflad I added the call and squashed the commits. (But, I find I get 100% pass on tests that are never called. 👍 )

**Comment by bflad** [comment]
> I find I get 100% pass on tests that are never called. 👍 

You should get this made into a poster, I would buy it! 😄

**Comment by YakDriver** [comment]
@bflad This is phenomenal! Thanks for your help and merge. When I get those t-shirts made, you'll get the first.

**Comment by bflad** [comment]
This has been released in [version 1.18.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1180-may-10-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Pull request merged**